metrics/metrics.go

```diff
@@ -0,0 +1,37 @@
+// Package metrics provides an extensible framework to instrument your
+// application. All metrics are safe for concurrent use. Considerable design
+// influence has been taken from https://github.com/codahale/metrics and
+// https://prometheus.io.
+package metrics
+
+// Counter is a monotonically-increasing, unsigned, 64-bit integer used to
+// capture the number of times an event has occurred. By tracking the deltas
+// between measurements of a counter over intervals of time, an aggregation
+// layer can derive rates, acceleration, etc.
+type Counter interface {
+	With(Field) Counter
+	Add(delta uint64)
+}
+
+// Gauge captures instantaneous measurements of something using signed, 64-bit
+// integers. The value does not need to be monotonic.
+type Gauge interface {
```

Comment:
At google I had precisely the opposite experience! Weird.

I agree that it's not difficult to wrap this, but again I've found that the common case for gauges is that there's already a source of ground truth in your app code somewhere, and you just need to expose it whenever the monitoring heartbeat happens. So a callback is a better fit. Compare this with a counter, of course, where you mainly interact with the counter when the counted thing has just occurred.

Could you humor me with an example or two of your gauge use case?